daemon/policy.go

```diff
@@ -302,6 +304,79 @@ func (d *Daemon) PolicyAdd(rules policyAPI.Rules, opts *AddOptions) (newRev uint
 	return newRev, nil
 }
 
+type policyAddQueueOptions struct {
+	rules   policyAPI.Rules
+	opts    *AddOptions
+	context *policyQueueContext
+	reFetch bool
+}
+
+type policyQueueContext struct {
+	policyRevision uint64
+	err            error
+	done           chan bool
+	totalTime      spanstat.SpanStat
+	queueTime      spanstat.SpanStat
+}
+
+var policyAddQueueRequests = make(chan policyAddQueueOptions, 100)
```

Comment:
Why isn't this channel part of the policy repository?

Reply:
umm, I thought all the day why this channel need to be in the policy repo but I can't fully see any reason. 

We want to avoid the races in PolicyAdd before policyRepo so I'm not sure why add a channel in the policy repo will fix this. 

Regards

Reply:
@aanm We can move this into the policy package as we move the majority of `daemon/policy.go` into the policy package.